_posts/2017-09-05-planning-for-7.0.md

```diff
@@ -0,0 +1,610 @@
+Output
+
+```js
+class Foo {
+  constructor() {
+    this.constructor;
+  }
+
+  test() {
+    void 0;
+  }
+}
+```
+
+## 🚀 New Feature
+
+### `.babelrc.js`
+
+> [babel/babel#4630](https://github.com/babel/babel/issues/4630)
+
+`*.js` configuration files are fairly common in the JavaScript ecosystem. ESlint and Webpack both allow for `.eslintrc.js` and `webpack.config.js` configuration files, respectively.
+
+Writing configuration files in JavaScript allows for dynamic configuration, making it possible to write a single configuration file that can adapt to different environments programmatically.
+
+```js
+var env = process.env.BABEL_ENV || process.env.NODE_ENV;
+var plugins = [];
+if (env === 'production') {
+  plugins.push.apply(plugins, ["a-super-cool-babel-plugin"]);
+}
+module.exports = { plugins };
+```
+
+```js
+var env = process.env.BABEL_ENV || process.env.NODE_ENV;
+module.exports = {
+  plugins: [
+    env === 'production' && "another-super-cool-babel-plugin"
+  ].filter(Boolean)
+};
+```
+
+This was previously done through the `env` configuration option, which is now deprecated. See [below](#deprecate-the-env-option-in-babelrc) for more details.
+
+### Typescript
+
+You can now use `babel-preset-typescript` to allow Babel to strip types similar to how `babel-preset-flow` works!
+
+```json
+{
+  "presets": ["typescript"]
+}
+```
+
+We're working on a guide for how to setup Typescript and Babel with the Typescript team, which should be finished before the official 7.0 release.
+
+## 😎 Other Breaking Changes
+
+### Removed `babel-preset-flow` from `babel-preset-react`
+
+This was important because we got a lot of complaints from users that weren't using any types/flow that they ended up writing invalid JS but there was no syntax error because they used the react preset.
+
+Also we have the typescript preset now, so it didn't make sense to include `flow` in the react preset itself anymore.
+
+### Integrations
+
+Packages like `grunt-babel`, `gulp-babel`, `rollup-plugin-babel`, etc all used to have `babel-core` as a dependency.
+
+After v7, we plan to move `babel-core` to be a peerDependency like `babel-loader` has. This lets all these packages not have to bump major versions when the `babel-core` API hasn't changed. Thus they are already published as `7.0.0` since we don't expect any further changes to those packages.
+
+## Meta
+
+### [#5218](https://github.com/babel/babel/pull/5218) Remove `babel-runtime` from our own Babel dependencies
+
+Babel itself doesn't have that many external dependencies, but in 6.x [each package has a dependency on `babel-runtime`](https://github.com/babel/babel/blob/958f72ddc28e2f5d02adf44eadd2b1265dd0fa4d/packages/babel-plugin-transform-es2015-arrow-functions/package.json#L12) so that built-ins like Symbol, Map, Set, and others are available without needing a polyfill. By changing the minimum supported version of Node to v4 (where those built-ins are supported natively), we can drop the dependency entirely.
```

Comment:
This would also reduce the confusion around `babel-runtime` quite a bit.

Reply:
FYI this is only for babel itself, not for users though so they'd still need it.